server/api/crud/files.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import mimetypes
+from http import HTTPStatus
+
+import mlrun.utils.singleton
+import server.api.api.utils
+from mlrun import store_manager
+from mlrun.errors import err_to_str
+from mlrun.utils import logger
+
+
+class Files(
+    metaclass=mlrun.utils.singleton.Singleton,
+):
+    def get_filestat_with_project_secrets(
```

Comment:
what's the difference between both `get_filestat_with_project_secrets` and `get_filestat` if they both use project secrets? seems redundant. use `get_filestat` only